pkg/apis/componentconfig/types.go

```diff
@@ -62,3 +62,213 @@ const (
 	ProxyModeUserspace ProxyMode = "userspace"
 	ProxyModeIPTables  ProxyMode = "iptables"
 )
+
+type KubeletConfiguration struct {
+	// config is the path to the config file or directory of files
+	Config string `json:"config"`
```

Comment:
If we are going to curate this in the future then no point in spending time making them sorted if you have not already done so.  Please add a TODO to better document the logical groupings and make them explicit.